### PR TITLE
OOP school library: decorate a class

### PR DESCRIPTION
**Pull request by nahnahdev**
## Project requirements

- [ ] Check the usage of [Decorator pattern in Ruby](https://refactoring.guru/design-patterns/decorator/ruby/example#example-0)
- [ ]  Use two decorators in order to capitalize and trim people's names.
- [ ] Follow the steps to get instructions for implementing the Decorator design pattern for this case.

```
- Interface
- Turn your Person class to Nameable
- Prepare base Decorator
- Prepare CapitalizeDecorator and TrimmerDecorator
```

### CODE REVIEW

File: person.rb [hunk 1]
```diff
@@ -7,12 +7,24 @@
...
-class Person
+# Interface
+# Create a class Nameable.
+# Implement a method called correct_name that will raise a NotImplementedError.
+
+class Nameable
+  def correct_name
+    raise NotImplementedError, 'Please implement the Nameable#correct_name method'
+  end
+end
```
Comment (TedLivist) [inline person.rb]: - [ ] Based on ruby conventions each class should be in its own file. Kindly move this class to its own file, then import it instead, for example:

  ```
  require_relative 'nameable.rb'
  ```

File: person.rb [hunk 2]
```diff
@@ -7,12 +7,24 @@
...
+
+# Make sure that your Person class inherits from Nameable
+class Person < Nameable
   attr_accessor :name, :age
   attr_reader :id
 
   def
       initialize(age, name = 'Unknown', parent_permission: true)
```
Comment (TedLivist) [inline person.rb]: - [ ] I think this should be on the same line.

  ```
  def initialize(...)
  ```

File: person.rb [hunk 3]
```diff
@@ -23,9 +35,59 @@ def can_use_services?
...
+class BaseDecorator < Nameable
+  def initialize(nameable)
+    super()
+    @nameable = nameable
+  end
+
+  def correct_name
+    @nameable.correct_name
+  end
+end
```
Comment (TedLivist) [inline person.rb]: - [ ] Kindly move this class to its own file.

File: person.rb [hunk 4]
```diff
@@ -23,9 +35,59 @@ def can_use_services?
...
+
+# Prepare CapitalizeDecorator and TrimmerDecorator
+# For the CapitalizeDecorator:
+# Create a class that inherits from the base Decorator class.
+class CapitalizeDecorator < BaseDecorator
+  # Implement a method correct_name that capitalizes the output of @nameable.correct_name.
+  def correct_name
+    @nameable.correct_name.capitalize
+  end
+end
```
Comment (TedLivist) [inline person.rb]: - [ ] Kindly move this class to its own file.

File: person.rb [hunk 5]
```diff
@@ -23,9 +35,59 @@ def can_use_services?
...
+# Create a class that inherits from the base Decorator class.
+class TrimmerDecorator < BaseDecorator
+  # Implement a method correct_name that makes sure that the output of @nameable.correct_name
+  #  has a maximum of 10 characters. If it's longer it should trim the word.
+  def correct_name
+    correct_name = @nameable.correct_name
+    correct_name = correct_name.slice(0, 10) if correct_name.length > 10
+    correct_name
+  end
+end
```
Comment (TedLivist) [inline person.rb]: - [ ] Kindly move this class to its own file.

**Review by TedLivist — CHANGES_REQUESTED** [review]
## STATUS: CHANGES REQUIRED ♻️ ♻️

Hi @NahnahAJ,

Good job so far!
✔️ Good documentation
✔️ Implementation of class inheritance
However, there are some issues that you still need to work on to go to the next project but you are almost there!

## Required Changes ♻️
_Check the comments under the review._

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear. And please, remember to tag me in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Review by m05e5 — APPROVED** [review]
Hi @NahnahAJ , 

Great Job 😄  . Your project is complete!  There is nothing else to say other than... it's time to merge it :shipit: 
Congratulations! 🎉 
goodjob

## Highlights
✔️ your linters are all ok
✔️ Descriptive PR 

## Optional suggestions


_Every comment with the [OPTIONAL] prefix won't stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better. Some of them were simply missed by the previous reviewer and addressing them will really improve your application._

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

_**m05e5**_

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Pull request merged**